src/main/java/org/opentripplanner/middleware/docs/PublicApiDocGenerator.java

```diff
@@ -0,0 +1,416 @@
+                //     type: array
+                //     items:
+                //       $ref: "#/definitions/MonitoredTrip"
+                schemaNode.remove(REF_FIELD);
+                schemaNode
+                    .put("type", "array")
+                    .set("items", JsonNodeFactory.instance.objectNode()
+                        .put(REF_FIELD, DEFINITIONS_REF + baseType));
+
+                // Delete the autogenerated array definition.
+                definitionsNode.remove(baseType + ARRAY_MARKER);
+            }
+        }
+    }
+
+    /**
+     * Generate missing types not created by spark-swagger.
+     * Note: This method might not end up creating types if there are no missing types.
+     */
+    private void generateMissingTypes() {
+        // Find the types from all $ref entries.
+        for (JsonNode node: rootNode.findValues(REF_FIELD)) {
+            if (node.asText().startsWith(DEFINITIONS_REF)) {
+                String baseType = node.asText().substring(DEFINITIONS_REF.length());
+
+                // Add type if it is not defined.
+                if (definitionsNode.get(baseType) == null) {
+                    JsonNode typeDefinition = JsonNodeFactory.instance.objectNode()
+                        .put("type", "object")
+                        .put("description", "autogenerated");
+
+                    definitionsNode.set(baseType, typeDefinition);
+                }
+            }
+        }
+    }
+
+    /**
+     * Removes unused types (not referenced in a $ref entry anywhere) from the definitions section.
+     */
+    private void removeUnusedTypes() {
+        int unusedTypeCount;
+        // Execute until there are no unused types.
+        do {
+            // Find return type references anywhere (only store one instance of each).
+            HashSet<String> typeReferences = new HashSet<>();
+            for (JsonNode refNode : rootNode.findValues(REF_FIELD)) {
+                // The reference is typically in the form "#/definitions/TypeName".
+                // Extract the "raw" type (TypeName in example above) from the definition string.
+                String typeName = refNode.asText().substring(DEFINITIONS_REF.length());
+                typeReferences.add(typeName);
+            }
+
+            // List types from the definitions node that don't have a reference from the list above.
+            // Then remove unused types from the definitions node.
+            List<String> typesToRemove = getFieldNames(definitionsNode, type -> !typeReferences.contains(type));
+            definitionsNode.remove(typesToRemove);
+
+            unusedTypeCount = typesToRemove.size();
+        } while(unusedTypeCount > 0);
+    }
+
+    /**
+     * Insert the definition for {@link:AbstractUser}
+     * and add fields (with description) from AbstractUser to OtpUser.
+     * (spark-swagger ignores parent classes when generating swagger.)
+     * TODO: Find a better way to include parent classes.
```

Comment:
Might be good to mention that this skips the other user child classes because they are not "public-facing".

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/59/commits/b327a799b261096bd67fcaf0d680d92b1090fc2a